### PR TITLE
translate deepspeed.md to chinese

### PR DESCRIPTION
**Pull request by jiaqiw09**
# What does this PR do?

Part of  #26803 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests? _not necessary_

## Who can review?
@stevhliu

### CODE REVIEW

**Review by stevhliu — APPROVED** [review]
LGTM! My Chinese isn't that great (its pretty basic 😅 ) so maybe @statelesshz, would you be interested in helping double check the translation?

**Comment by statelesshz** [comment]
> LGTM! My Chinese isn't that great (its pretty basic 😅 ) so maybe @statelesshz, would you be interested in helping double check the translation?

My Pleasure!

File: docs/source/zh/main_classes/deepspeed.md [hunk 1]
```diff
@@ -0,0 +1,2101 @@
...
+DeepSpeed ZeRO-2主要用于训练，因为它的特性对推理没有用处。
+
+DeepSpeed ZeRO-3也可以用于推理，因为它允许将单个GPU无法加载的大模型加载到多个GPU上。
+
+🤗 Transformers通过以下两种方式集成了[DeepSpeed](https://github.com/microsoft/DeepSpeed)：
+
+1. 通过[`Trainer`]集成核心的DeepSpeed功能。这是一种“一切为您完成”类型的集成 - 只需提供您的自定义配置文件或使用我们的模板配置文件，您就不需要做其他事情。本文档的大部分内容都集中在这个功能上。
+2. 如果您不使用[`Trainer`]并希望在自己的Trainer中集成DeepSpeed，那么像`from_pretrained`和`from_config`这样的核心功能函数将包括ZeRO stage 3及以上的DeepSpeed的基础部分，如`zero.Init`。要利用此功能，请阅读有关[非Trainer DeepSpeed集成](#nontrainer-deepspeed-integration)的文档。
+
+集成的内容是什么：
```
Comment (statelesshz) [inline docs/source/zh/main_classes/deepspeed.md:39]:
```suggestion
集成的内容：
```

File: docs/source/zh/main_classes/deepspeed.md [hunk 2]
```diff
@@ -0,0 +1,2101 @@
...
+
+训练：
+
+1. DeepSpeed ZeRO训练支持完整的ZeRO stages 1、2和3，以及ZeRO-Infinity（CPU和NVMe offload）。
+
+推理：
+
+1. DeepSpeed ZeRO推理支持使用ZeRO-Infinity的ZeRO stage 3。它使用与训练相同的ZeRO协议，但不使用优化器和学习率调度器，只与stage 3相关。更多详细信息请参阅：[zero-inference](#zero-inference)。
```
Comment (statelesshz) [inline docs/source/zh/main_classes/deepspeed.md:47]:
```suggestion
1. DeepSpeed ZeRO推理支持ZeRO stage 3和ZeRO-Infinity。它使用与训练相同的ZeRO协议，但不使用优化器和学习率调度器，只有stage 3与推理相关。更多详细信息请参阅：[zero-inference](#zero-inference)。
```

**Review by ji-huazhong — CHANGES_REQUESTED** [review]

**Comment by jiaqiw09** [comment]
@statelesshz thanks for your reviw.

and @stevhliu would you mind having a check and finish the merge. I think I will update this file oneday I learn more about deepspeed.

Best

**Comment by stevhliu** [comment]
Great job and thanks again for tackling such a big and dense doc! 👏

**Pull request merged**